Tests/Realm.Tests/Database/AsymmetricObjectTests.cs

```diff
@@ -0,0 +1,382 @@
+
+                flxConfig.PopulateInitialSubscriptions = (realm) =>
+                {
+                    var query = realm.All<PrimaryKeyInt32Object>().Where(n => n.Id == id);
+                    realm.Subscriptions.Add(query);
+                };
+
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+
+                Assert.DoesNotThrow(() =>
+                {
+                    realm.Write(() =>
+                    {
+                        realm.Add(new BasicAsymmetricObject
+                        {
+                            PartitionLike = partitionLike
+                        });
+
+                        realm.Add(new PrimaryKeyInt32Object
+                        {
+                            Id = id
+                        });
+                    });
+                });
+            });
+        }
+
+        [Test, NUnit.Framework.Explicit("Once Daniel Tabacaru's work is done on \"error actions\" this will be an \"application bug\" action: https://github.com/10gen/baas/blob/9f32d54aa79aff6dfb36a6c07742594e38b07441/realm/sync/protocol/protocol_errors.go#L439")]
+        public void AsymmetricObjectInPbs_Throws()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var config = await GetIntegrationConfigAsync();
+                config.Schema = new[] { typeof(BasicAsymmetricObject) };
+                var tcs = new TaskCompletionSource<object>();
+
+                config.OnSessionError = (session, error) =>
+                {
+                    Assert.That(error, Is.InstanceOf<SessionException>());
+
+                    // Assert.That(error.ErrorCode, Is.EqualTo(ErrorCode.ApplicationBug));
+                };
+
+                using var realm = await GetRealmAsync(config);
+
+                await tcs.Task;
+            });
+        }
+
+        [Test]
+        public void AsymmetricObjectInLocalRealm_Throws()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var config = (RealmConfiguration)RealmConfiguration.DefaultConfiguration;
+                config.Schema = new[] { typeof(BasicAsymmetricObject) };
+
+                Exception ex = null;
+                try
+                {
+                    using var realm = await GetRealmAsync(config);
+                }
+                catch (Exception e)
+                {
+                    ex = e;
+                }
+
+                Assert.That(ex, Is.InstanceOf<RealmSchemaValidationException>());
+                Assert.That(ex.Message.Contains($"Asymmetric table \'{nameof(BasicAsymmetricObject)}\'"), Is.True);
+            });
+        }
+
+        private static Task<T[]> GetObjFromRemoteThroughMongoClient<T>(FlexibleSyncConfiguration config, string remoteFieldName, BsonValue fieldValue, string mongoClientCondition = MongoClientCondition.Equality)
```

Comment:
```suggestion
        private static Task<T[]> GetObjFromRemoteThroughMongoClient<T>(User user, string remoteFieldName, BsonValue fieldValue, string mongoClientCondition = MongoClientCondition.Equality)
```

I thought it's quite confusing we're passing a config here.